src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -311,20 +317,45 @@ else if (SiteUtil.JAVADOC_TOKENS.equals(propertyName)) {
         }
         else {
             final String type = SiteUtil.getType(field, propertyName, currentModuleName, instance);
-            final String relativePathToPropertyTypes =
-                    SiteUtil.getLinkToDocument(currentModuleName, "property_types.xml");
-            final String escapedType = type
-                    .replace("[", ".5B")
-                    .replace("]", ".5D");
-            final String url =
-                    String.format(Locale.ROOT, "%s#%s", relativePathToPropertyTypes, escapedType);
-            sink.link(url);
-            sink.text(type);
-            sink.link_();
+            if ("subset of tokens TokenTypes".equals(type)) {
```

Comment:
Truthfully, I question this kind of matching.

We are losing type safety by not matching on a type like `XdocsPagesTest` did, by requiring a match on a custom string. This may be a limitation of how we are coding this macro, but custom strings are usually not a good idea. It is easy to misspell or typo something and go unnoticed.

I also question where these strings are coming from, and why don't we match on defined public string from there, similar to what we did with `PROPERTY_TYPES_XML`. Something like `if (type.equals(TOK_STR_1 + TOK_STR_2))`.

Reply:
> Truthfully, I question this kind of matching.
> 
> We are losing type safety by not matching on a type like `XdocsPagesTest` did, by requiring a match on a custom string. This may be a limitation of how we are coding this macro, but custom strings are usually not a good idea. It is easy to misspell or typo something and go unnoticed.
> 
> I also question where these strings are coming from, and why don't we match on defined public string from there, similar to what we did with `PROPERTY_TYPES_XML`. Something like `if (type.equals(TOK_STR_1 + TOK_STR_2))`.

I agree with type-safety issue. I personally didn't like the string literals but i assume since it was more of a hack, and a sustainable solution is waiting, i didn't create constants. However, in my coming push, i have refactored the code as such:
 `if (type.equals(SUBSET_TOKENS + TOKENS_PROPERTY)) {`
 
 I created a constant for the same and used an existing one. Please let me know your feedback.

Reply:
> PROPERTY_TYPES_XM

Or, do you want me to extract 'subset of tokens' from somewhere inside the code?

Reply:
from `TOKEN_ARRAY("subset of tokens TokenTypes");`

being used here:
```
  @XdocsPropertyType(PropertyType.TOKEN_ARRAY)
    private int[] limitedTokens = CommonUtil.EMPTY_INT_ARRAY;
```

We see the same in xdocs. 

And, all the below are using `TOKEN_ARRAY`
```
private static final Set<String> NON_BASE_TOKEN_PROPERTIES = Collections.unmodifiableSet(
      Arrays.stream(new String[] {
          "AtclauseOrder - target",
          "DescendantToken - limitedTokens",
          "IllegalType - memberModifiers",
          "MagicNumber - constantWaiverParentToken",
          "MultipleStringLiterals - ignoreOccurrenceContext",
      }).collect(Collectors.toSet()));
```

Reply:
We see the same issue in migration of all below files. Maybe, this hack for this PR will help migration of the above checks mentioend.

Reply:
And, in my code flow, it directly enters else block with value as `"subset of tokens TokenTypes"`
  ```
private static void writePropertyTypeCell(Sink sink, String propertyName,
                                              Field field, Object instance)
            throws MacroExecutionException {
        sink.rawText(INDENT_LEVEL_14);
        sink.tableCell();

        if (SiteUtil.TOKENS.equals(propertyName)) {
            final AbstractCheck check = (AbstractCheck) instance;
            if (check.getRequiredTokens().length == 0
                    && Arrays.equals(check.getAcceptableTokens(), TokenUtil.getAllTokenIds())) {
                sink.text("set of any supported");
                writeLink(sink);
            }
            else {
                final List<String> configurableTokens = SiteUtil
                        .getDifference(check.getAcceptableTokens(),
                                check.getRequiredTokens())
                        .stream()
                        .map(TokenUtil::getTokenName)
                        .collect(Collectors.toList());
                sink.text(SUBSET_TOKENS);

                writeTokensList(sink, configurableTokens, SiteUtil.PATH_TO_TOKEN_TYPES);
            }
        }
        else if (SiteUtil.JAVADOC_TOKENS.equals(propertyName)) {
            final AbstractJavadocCheck check = (AbstractJavadocCheck) instance;
            final List<String> configurableTokens = SiteUtil
                    .getDifference(check.getAcceptableJavadocTokens(),
                            check.getRequiredJavadocTokens())
                    .stream()
                    .map(JavadocUtil::getTokenName)
                    .collect(Collectors.toList());
            sink.text("subset of javadoc tokens");
            writeTokensList(sink, configurableTokens, SiteUtil.PATH_TO_JAVADOC_TOKEN_TYPES);
        }
        else {
            final String type = SiteUtil.getType(field, propertyName, currentModuleName, instance);
            if ((SUBSET_TOKENS + " " + TOKEN_TYPES_PROPERTY).equals(type)) {
                processLinkForTokenTypes(sink);
            }
            else {
                final String relativePathToPropertyTypes =
                        SiteUtil.getLinkToDocument(currentModuleName, PROPERTY_TYPES_XML);
                final String escapedType = type
                        .replace("[", ".5B")
                        .replace("]", ".5D");
                final String url =
                        String.format(Locale.ROOT, URL_F, relativePathToPropertyTypes, escapedType);
                sink.link(url);
                sink.text(type);
                sink.link_();
            }
        }
        sink.tableCell_();
    }
```
    
   And, if we ignore the  `if ((SUBSET_TOKENS + " " + TOKEN_TYPES_PROPERTY).equals(type)) {` added by me, it goes on to write the xdocs with link followed by type which is nothing but value held by `TOKENS_ARRAY`. So, if i don't add any hack and just proceed with normal migration, you see something like this:
`<td><a href="../../property_types.html#subset of tokens TokenTypes">subset of tokens TokenTypes</a></td>`

Reply:
And, that leads to linkcheck error in CI. So, either way, it is broken.

Reply:
And, if migration is a priority, we should proceed with this PR. Otherwise, in whatever little time, i have understood the code, it needs to be re-written to accomodate all edge cases we have been encountering with properties migration. So, if @rnveach, if you agree, we can proceed with the PR and i can start working with your guiding light on how to get rid of hacks so far.

Reply:
```suggestion
            if (PropertyType.TOKEN_ARRAY.getDescription().equals(type)) {
```

This resolves the type safety and usage of raw text concern.